internal/controller/grpc_logger.go

```diff
@@ -58,13 +58,6 @@ func (b *BufferedGrpcWriteSyncer) Close() error {
 	b.mutex.Lock()
 	defer b.mutex.Unlock()
 	b.flush()
-	// Close the channel if not already closed.
```

Comment:
Why did you delete this?
The code now unconditionally closes the channel, which panics if done more than once.
The code you deleted was explicitly added to make `Close()` idempotent.